From 307abbfe89117a95dcceb5ee33c54fb70f668928 Mon Sep 17 00:00:00 2001
From: David Bauer <mail@david-bauer.net>
Date: Thu, 16 Mar 2023 19:21:04 +0530
Subject: [PATCH 106/500] OpenWrt:
 205-kconfig-abort-configuration-on-unset-symbol.patch

kconfig: abort configuration on unset symbol

When a target configuration has unset Kconfig symbols, the build will
fail when OpenWrt is compiled with V=s and stdin is connected to a tty.

In case OpenWrt is compiled without either of these preconditions, the
build will succeed with the symbols in question being unset.

Modify the kernel configuration in a way it fails on unset symbols
regardless of the aforementioned preconditions.

Signed-off-by: David Bauer <mail@david-bauer.net>

Change-Id: If3f6f7c3a2af97b4fc32d5f397232e5aa19f4eca
Signed-off-by: Ram Chandra Jangir <quic_rjangir@quicinc.com>
(cherry picked from commit 865a1598214dcc79f5ff299f1d723e9b545ab22a)
---
 scripts/kconfig/conf.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/scripts/kconfig/conf.c b/scripts/kconfig/conf.c
index 33d19e419908..b2092a91d91f 100644
--- a/scripts/kconfig/conf.c
+++ b/scripts/kconfig/conf.c
@@ -338,6 +338,9 @@ static int conf_askvalue(struct symbol *sym, const char *def)
 		}
 		/* fall through */
 	default:
+		if (!tty_stdio && getenv("FAIL_ON_UNCONFIGURED")) {
+			exit(1);
+		}
 		fflush(stdout);
 		xfgets(line, sizeof(line), stdin);
 		break;
@@ -520,6 +523,9 @@ static int conf_choice(struct menu *menu)
 			}
 			/* fall through */
 		case oldaskconfig:
+			if (!tty_stdio && getenv("FAIL_ON_UNCONFIGURED")) {
+				exit(1);
+			}
 			fflush(stdout);
 			xfgets(line, sizeof(line), stdin);
 			strip(line);
-- 
2.34.1

